nntrainer/src/neuralnet.cpp

```diff
@@ -164,6 +165,13 @@ int NeuralNetwork::loadDatasetConfig(void *_ini) {
 }
 
 int NeuralNetwork::loadFromConfig() {
+  if (loadedFromConfig == true) {
+    ml_loge("cannnot do loadFromConfig twice");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  loadedFromConfig = true;
```

Comment:
this should be set at the end of function.
this will remain true even if loadFromConfig fails.

Reply:
Done

Reply:
This adds more issues to a function which already has more errors.
Keep loadFromConfig to the end. 
Reset can be done separately in a new PR, but setting it at start creates more issues where this value also has to be reset.
In general, such values as configured/loaded etc should be set at the end of the call.

Reply:
> it at start creates more issues

Can you elaborate on this? Since the function is not atomic, it has to be called only once fail or succeed. If you allow it to call again when fail, it will be running without exposing bugs.

Reply:
If it fails once with some incorrect config, and then called again with correct config, it will fail as loadFromConfig is not reset correctly when failed.

Reply:
> If it fails once with some incorrect config, and then called again with correct config, it will fail as loadFromConfig is not reset correctly when failed.

If it is, it ought to go bottom. The problem is it isn't as #382 states

Reply:
Yes. Now its good.